solr/core/src/java/org/apache/solr/servlet/PrometheusStatzServlet.java

```diff
@@ -0,0 +1,151 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.solr.servlet;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import java.io.OutputStreamWriter;
+import java.io.PrintWriter;
+import java.io.Writer;
+import java.lang.management.GarbageCollectorMXBean;
+import java.lang.management.ManagementFactory;
+import java.lang.management.MemoryMXBean;
+import java.lang.management.MemoryUsage;
+import java.nio.charset.StandardCharsets;
+import java.util.LinkedHashMap;
+import java.util.Map;
+
+import com.sun.management.UnixOperatingSystemMXBean;
+import org.apache.commons.io.IOUtils;
+import org.apache.commons.io.output.CloseShieldOutputStream;
+ /**
+ * FullStory: a simple servlet to produce a few stats formatted for Prometheus.
+ */
+public final class PrometheusStatzServlet extends BaseSolrServlet {
+   public static void writePromDoc(PrintWriter writer, String name, String purpose, String type) {
+     writer.printf("# HELP %s %s", name, purpose);
+     writer.println();
+     writer.printf("# TYPE %s %s", name, type);
+     writer.println();
+   }
+   @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+     Writer out = null;
+    try {
+      response.setCharacterEncoding("UTF-8");
+      response.setContentType("application/json");
+       // Protect container owned streams from being closed by us, see SOLR-8933
+      out = new OutputStreamWriter(new CloseShieldOutputStream(response.getOutputStream()), StandardCharsets.UTF_8);
+      writeStats(new PrintWriter(out));
+    } finally {
+      IOUtils.closeQuietly(out);
+    }
+  }
```

Comment:
You should be able to write something like:

```java
try (Writer out = new OutputStreamWriter(new CloseShieldOutputStream(response.getOutputStream()), StandardCharsets.UTF_8)) {
 ....
}
```

https://docs.oracle.com/javase/tutorial/essential/exceptions/tryResourceClose.html